src/features/courses/course-controller.ts

```diff
@@ -959,8 +1022,46 @@ class CourseController {
                 studentWorkbook: null
             };
         }
-        const question: CourseWWTopicQuestion = await studentGrade.getQuestion();
-        const topic: CourseTopicContent = await question.getTopic();
+        const question: (CourseWWTopicQuestion) = await studentGrade.getQuestion(
+            {
+                include: [{
+                        model: StudentTopicQuestionOverride,
+                        as: 'studentTopicQuestionOverride',
+                        attributes: ['userId', 'maxAttempts'],
+                        required: false,
+                        where: {
+                            active: true,
+                            userId: options.userId
+                        }
+                }]
+            }
+        );
+        
+        
+        const topic: CourseTopicContent = await question.getTopic(            {
+            include: [{
+                model: StudentTopicOverride,
+                as: 'studentTopicOverride',
+                attributes: ['userId', 'startDate', 'endDate', 'deadDate'],
+                required: false,
+                where: {
+                    active: true,
+                    userId: options.userId
+                }
+            }]
+        });
+        
+        if ((topic as any)?.studentTopicOverride?.length === 1) {
```

Comment:
Not Nit
Linter should have yelled at you for using any
You should add studentTopicOverride to the topic class, then you don't need to cast
Also if you do break linting rules you should add the ignore comment and add a comment as to why you ignored the rule

Reply:
Like you did below 🤣 the below one I can see being a pain and can understand pushing that off because of the `assign`

Reply:
Thanks, I should've checked how you were handling it over there.